Added WalletSelector Scene & Wallet selector vm in connection proposa…

gemdev111 opened the pull request:
resolve #17 

Added ui selector for wallets ( proposal scene ) 
Updates wallet-connect logic to support selected wallet in proposal

images:
<table>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/562dce66-18ab-4f34-8c7b-656b4134f3b7" style="width: 45%;" /></td>
    <td><img src="https://github.com/user-attachments/assets/71e6336f-6fed-408e-96e5-ff2fb0ac0e42" style="width: 45%;" /></td>
  </tr>
</table>

gemdev111 committed:
Added WalletSelector Scene & Wallet selector vm in connection proposal scene
resolve #17 ( in progress )

gemcoder21 commented on Gem/Connections/Scenes/ConnectionProposalScene.swift:
```diff
@@ -25,29 +29,37 @@ struct ConnectionProposalScene: View {
+                    NavigationLink(value: Scenes.WalletsSelector()) {
+                        ListItemView(
+                            title: Localized.Common.wallet,
+                            subtitle: model.selectedWalletName
```
```suggestion
                            subtitle: model.walletName
```

gemcoder21 commented on Gem/Connections/Scenes/WalletSelectorScene.swift:
```diff
@@ -0,0 +1,53 @@
+import Components
+import Style
+
+struct WalletsSelectorScene: View {
```
```suggestion
struct SelectWalletScene: View {
```

gemcoder21 commented on Gem/Connections/ViewModels/WalletSelectorViewModel.swift:
```diff
@@ -0,0 +1,29 @@
+import Foundation
+import Primitives
+
+struct WalletSelectorViewModel {
```
```suggestion
struct SelectWalletViewModel {
```

gemcoder21 commented on Gem/Core/Types/Scenes.swift:
```diff
@@ -19,6 +19,7 @@ struct Scenes {
     struct Validators: Hashable {}
     struct Stake: Hashable {}
     struct WalletConnect: Hashable {}
+    struct WalletsSelector: Hashable {}
```
```suggestion
    struct SelectWallet: Hashable {}
```

gemdev111 committed:
Removed Combine
WalletConnector code strucutre refactored, updated from combine publishers to async stream

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Adjusted logic to support pairing with any wallet you select in connection proposal scene

gemdev111 committed:
Merge branch 'main' into wallet-connect-picker

gemdev111 committed:
Merge branch 'main' into wallet-connect-picker

gemdev111 committed:
minor update, removed unnecessary check

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Updates wallet connect transaction sending by adding WCTransferData with wc wallet

gemdev111 committed:
Merge branch 'main' into wallet-connect-picker

gemdev111 committed:
Updates app tint(navigation bar items from blue to dark/white)

gemdev111 committed:
adjustments absed on @gemcoder21 review

gemdev111 committed:
Merge branch 'main' into wallet-connect-picker
